01_fib/nicholascross/main.go

```diff
@@ -0,0 +1,41 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+func main() {
+	Fib(7)
+}
+
+func Fib(n int) {
+	if n == 0 {
+		return
+	}
+
+	sign := sign(n)
+	sequenceLength := sign * n
+
+	lastNumber := 0
+	currentNumber := 1
+	fmt.Fprint(out, currentNumber)
+
+	for index := 0; index < sequenceLength-1; index++ {
+		nextNumber := lastNumber + currentNumber*sign
```

Comment:
there is a convention within the go community to use very short variable names within functions. When you are reviewing the context of the function then a variablename like `ln` instead of `lastName` allows for faster reading whilst still allowing you to parse the whole function. I would suggest you refactor for smaller var names.

Reply:
I am feeling compelled to dissent as this is completely counter to classic advice where variable name brevity should be avoided for the sake of clarity and readability.

If that is the convention then, I will update.

Are there any guiding principals for good variable names in go or is it just keep it short as possible?

Reply:
I generally follow (and I feel this is common convention within the community) a convention where variables (or consts) which are going to be long lived should have a descriptive name, as it may be referenced away from it's initial declaration in the code. Anything which is to be short lived should have a concise name, see this for further reference: [Why does go promote short names for variables?](https://www.quora.com/Why-does-Golang-promote-short-and-kind-of-meaningless-names-for-variables)

For variables which live in the context for a single loop like this, I would be updating 
`index -> i` 
`seqenceLength -> len` (len is a reserved word, so something similar... maybe `length`
`nextNumber -> next`
`lastNumber -> last` 
`currentNumber -> curr`